### PR TITLE
Update Javadoc of premium SDK for azure-resourcemanager-containerregistry

### PR DESCRIPTION
**Pull request by v-hongli1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/CheckNameAvailabilityResult.java [hunk 1]
```diff
@@ -9,12 +9,24 @@
...
+    /**
+     * Gets the reason why the user-provided name for the container registry could not be used.
+     *
+     * @return the reason why the user-provided name for the container registry could not be used
+     */
     String unavailabilityReason();
 
-    /** @return the error message that provides more detail for the reason why the name is not available */
+    /**
+     * Gets provides more detail for the reason why the name is not available.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/CheckNameAvailabilityResult.java:27]:
```suggestion
     * Gets the error message that provides more detail for the reason why the name is not available.
```

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java [hunk 2]
```diff
@@ -27,22 +27,46 @@ public interface Registry extends GroupableResource<ContainerRegistryManager, Re
     Updatable<Registry.Update>, SupportsListingPrivateLinkResource, SupportsListingPrivateEndpointConnection,
     SupportsUpdatingPrivateEndpointConnection {
 
-    /** @return the SKU of the container registry. */
+    /**
+     * Gets the SKU of the container registry..
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java:31]:
```suggestion
     * Gets the SKU of the container registry.
```

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java [hunk 3]
```diff
@@ -27,22 +27,46 @@ public interface Registry extends GroupableResource<ContainerRegistryManager, Re
...
+    /**
+     * Gets the creation date of the container registry.
+     *
+     * @return the creation date of the container registry in ISO8601 format
+     */
     OffsetDateTime creationDate();
 
-    /** @return the value that indicates whether the admin user is enabled */
+    /**
+     * Checks whether the value that indicates whether the admin user is enabled.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java:52]:
```suggestion
     * Checks whether the admin user is enabled.
```

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java [hunk 4]
```diff
@@ -85,27 +113,53 @@ public interface Registry extends GroupableResource<ContainerRegistryManager, Re
...
+    /**
+     * Checks whether zone redundancy is enabled for this container registry.
+     *
+     * @return Whether zone redundancy is enabled for this container registry
+     */
     boolean isZoneRedundancyEnabled();
 
-    /** @return list of host names that will serve data when isDedicatedDataEndpointsEnabled is true */
+    /**
+     * Gets list of host names that will serve data.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java:159]:
```suggestion
     * Gets list of host names that will serve data when isDedicatedDataEndpointsEnabled is true.
```

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/Registry.java]: Fixed by the [commit#45f4e5a](https://github.com/Azure/azure-sdk-for-java/pull/43517/commits/45f4e5a69f93dddddb9daa8b89a1fed4b85b7ea0)

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**